src/code/z_skelanime.c

```diff
@@ -1,176 +1,1675 @@
+
+/* The next functions are callbacks to loading animations */
+
+void func_800A3678(GlobalContext* globalCtx, AnimationEntryType0* entry) {
+    osRecvMesg(&entry->msgQueue, NULL, OS_MESG_BLOCK);
+}
+
+void func_800A36A4(GlobalContext* globalCtx, AnimationEntryType1* entry) {
+    s32 i;
+    Vec3s* dst;
+    Vec3s* src;
+    AnimationEntryRaw* genericEntry = (AnimationEntryRaw*)entry;
+
+    if ((genericEntry->raw[0] & D_8012A480) != 0) {
+        return;
+    }
+
+    for (dst = entry->dst, src = entry->src, i = 0; i < genericEntry->raw[1]; i++) {
+        *dst++ = *src++;
+    }
+}
+
+void func_800A3714(GlobalContext* globalCtx, AnimationEntryType2* entry) {
+    if ((entry->unk_00 & D_8012A480) == 0) {
+        func_800A2E70(entry->unk_01, entry->unk_04, entry->unk_04, entry->unk_08, entry->unk_0C);
+    }
+}
+
+void func_800A3770(GlobalContext* globalCtx, AnimationEntryType3* entry) {
+    s32 i;
+    Vec3s* dst;
+    Vec3s* src;
+    u8* index;
+    AnimationEntryRaw* rawEntry = (AnimationEntryRaw*)entry;
+
+    if ((rawEntry->raw[0] & D_8012A480) == 0) {
+        for (dst = entry->dst, src = entry->src, index = entry->index, i = 0; i < rawEntry->raw[1]; i++, dst++, src++) {
+            if (*index++) {
+                *dst = *src;
+            }
+        }
+    }
+}
+
+void func_800A37F0(GlobalContext* globalCtx, AnimationEntryType4* entry) {
+    s32 i;
+    Vec3s* dst;
+    Vec3s* src;
+    u8* index;
+    AnimationEntryRaw* rawEntry = (AnimationEntryRaw*)entry;
+
+    if ((rawEntry->raw[0] & D_8012A480) == 0) {
+        for (dst = entry->dst, src = entry->src, index = entry->index, i = 0; i < rawEntry->raw[1]; i++, dst++, src++) {
+            if (*index++ < 1U) {
+                *dst = *src;
+            }
+        }
+    }
+}
+
+void func_800A3874(GlobalContext* globalCtx, AnimationEntryType5* entry) {
+    char pad[4];
+    Actor* actor = entry->actor;
+    Vec3f pos;
+
+    func_800A54FC(entry->skelAnime, &pos, actor->shape.rot.y);
+    actor->posRot.pos.x += pos.x * actor->scale.x;
+    actor->posRot.pos.y += pos.y * actor->scale.y * entry->unk_08;
+    actor->posRot.pos.z += pos.z * actor->scale.z;
+}
+
+void func_800A390C(GlobalContext* globalCtx, AnimationContext* animationCtx) {
+    AnimationEntry* entry;
+
+    for (entry = animationCtx->entries; animationCtx->animationCount != 0; entry++, animationCtx->animationCount--) {
+        D_8012A484[entry->type](globalCtx, &entry->types);
+    }
+
+    D_801600B0 = 1;
+    D_8012A480 = 0;
+}
+
+void func_800A39AC(GlobalContext* globalCtx, SkelAnime* skelAnime, u32 limbHeaderSeg, u32* animation, s32 flags,
+                   Vec3s* actorDrawTbl, Vec3s* arg6, s32 limbBufCount) {
+    char pad[8];
+    SkelAnimeLimbHeader* limbHeader;
+    size_t allocSize;
+    s32 limbCount;
+    s32 headerCount;
+
+    limbHeader = SEGMENTED_TO_VIRTUAL(limbHeaderSeg);
+    headerCount = limbHeader->limbCount;
+    skelAnime->unk_34 = flags;
+    limbCount = (flags & 2) != 0 ? headerCount : 1;
```

Comment:
I suggest omitting the != 0 or == 0 when dealing with obvious boolean checks like that. Unless it doesn't match ofc.